### PR TITLE
Update sidekiq-unique-jobs to v7

### PR DESCRIPTION
**Pull request by BeckaL**
Bumping the version of sidekiq-unique jobs to the latest involved a major version change (to v7).

This change requires us to:
* explicitly specify the middleware; and
* rename a couple of the sidekiq options for locking

Additionally, we have decided to remove testing for unique behaviour in tests, as builds were becoming flakey. This is recommended in documentation for the gem (see [here](https://github.com/mhenrixon/sidekiq-unique-jobs#uniqueness)). Instead we can test that the sidekiq options provided in the workers are valid.

### CODE REVIEW

**Review by kevindew — COMMENTED** [review]
This is looking great - thanks for working through this.

I'd suggest a couple of modifications to your commit messages: 

For the first one: state that we are upgrading rather than trying and for the last one it'd be good to explain what the flakey build situation was (state from one test affecting later tests). We tend to find ourselves looking at past commit messages a lot to understand why things were done so it's helpful to future GDS devs to have a good degree of context. We do have some documentation too: https://gds-way.cloudapps.digital/standards/source-code.html#commit-messages

File: spec/workers/check_worker_spec.rb [hunk 1]
```diff
@@ -1,11 +1,6 @@
 require "rails_helper"
 
 RSpec.describe CheckWorker do
-  # SidekiqUniqueJobs recommends not testing the behaviour, testing this has previously has caused flakey builds
-  before do
-    SidekiqUniqueJobs.config.enabled = false
-  end
```
[the remark this answers is not in the document]
Comment (BeckaL) [reply spec/workers/check_worker_spec.rb]: No this was actually me before I read the documentation properly, this wasn't existing code, something I added and then removed :-)

[the remark this answers is not in the document]
Comment (kevindew) [reply spec/workers/check_worker_spec.rb]: Oh yes, I see it introduced in https://github.com/alphagov/link-checker-api/pull/509/commits/e31777df199a04e4e57dce909d108b096858ef7f. 

Do you have experience of editing git commits with an interactive rebase ([docs](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History))? if so it'd be good to remove that from the history before we merge in - but no biggie.

[the remark this answers is not in the document]
Comment (BeckaL) [reply spec/workers/check_worker_spec.rb]: yeah I think it is removed from the history now? I rebased to clean up all the interim commits where I was trying to make this version of unique jobs work.

[the remark this answers is not in the document]
Comment (kevindew) [reply spec/workers/check_worker_spec.rb]: It looks like it's still there and introduced in https://github.com/alphagov/link-checker-api/pull/509/commits/e495391235fc11431802ea415ac84f88e213c142

I'd suggest going doing `git rebase --interactive d36a48d5234c637a3b21582c4bb638873783fa80`

and mark e495391 as a commit to edit. Edit the files to remove the: 

```

  # SidekiqUniqueJobs recommends not testing the behaviour, testing this has previously has caused flakey builds
  before do
    SidekiqUniqueJobs.config.enabled = false
  end
```

blocks.

Then you can do `git add <path to changed files>` and `git rebase --continue` which should see it sorted.

[the remark this answers is not in the document]
Comment (BeckaL) [reply spec/workers/check_worker_spec.rb]: Oh I see sorry 🤦‍♀️ thought had squashed the introduction and removal into the same commit. Thanks!

[the remark this answers is not in the document]
Comment (BeckaL) [reply spec/workers/check_worker_spec.rb]: An embarrassing number of force pushes later... should all be cleaned up now, also moved removing the webook worker test to the commit that's about removing tests for unique worker, so hopefully a more helpful commit history now

[the remark this answers is not in the document]
Comment (kevindew) [reply spec/workers/check_worker_spec.rb]: Super - looks great now. And don't worry, we've all been there on the force pushes.

**Pull request merged**